framework/src/base/MooseApp.C

```diff
@@ -117,6 +118,7 @@ MooseApp::MooseApp(InputParameters parameters) :
     _start_time_set(false),
     _start_time(0.0),
     _global_time_offset(0.0),
+    _output_warehouse(*this, getParam<bool>("buffer_action_console_outputs")),
```

Comment:
For some reason, I don't like this option being passed in the constructor of the warehouse. I would rather see MooseApp call a method in the constructor:

```
_output_warehouse.bufferActionOutputsBeforeConstruction(getParam<bool>("buffer..."));
```